websites/S/Sonic Retro/presence.ts

```diff
@@ -0,0 +1,275 @@
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			startTimestamp: browsingTimestamp
+		},
+		{ pathname, search, hostname } = window.location;
+	switch (hostname) {
+		case "sonicretro.org": {
+			presenceData.smallImageKey = "home_512";
+			presenceData.smallImageText = "sonicretro.org";
+			if (search) {
+				presenceData.largeImageKey = "search";
+				if (search === "?s=")
+					presenceData.details = "Searching for something...";
+				else {
+					presenceData.details = `Searching for "${
+						document.querySelector("#main > div.archive-box > h1").textContent
+					}"`;
+				}
+			} else if (
+				((pathname === "/" || pathname === "/#") && !search) ||
+				(pathname.includes("/page/") && !search)
+			) {
+				presenceData.largeImageKey = "home";
+				presenceData.details = "Zooming through the main page";
+			} else if (pathname.includes("/category/")) {
+				presenceData.largeImageKey = "search";
+				presenceData.details = `Looking through ${
+					document.querySelector("#main > div.archive-box > h1").textContent
+				}`;
+			} else if (/\d{4}\/\d{2}\/(\d{2})*/.test(document.location.pathname)) {
+				presenceData.largeImageKey = "search";
+				presenceData.details = `Looking for posts from ${
+					document.title.split("-")[0]
+				}`;
+			} else if (pathname.includes("/wp-content/")) {
+				// viewing uploaded content; unnecesary to show in the presence
+				presenceData.largeImageKey = "logo_1024";
+				presenceData.details = "Viewing an unsupported page";
+			} else if (
+				/\/([0-9]+(\/[0-9]+)+)\/+((\w|-|\d)*)\//g.test(location.href)
+			) {
+				presenceData.largeImageKey = "reading";
+				presenceData.details = "Reading an article";
+				presenceData.state = `${document.title.split(" - Sonic Retro")[0]}`;
+				presenceData.buttons = [
+					{
+						label: "Read Article",
+						url: location.href
+					}
+				];
+			} else {
+				presenceData.largeImageKey = "logo_1024";
+				presenceData.details = "Viewing an unsupported page";
+			}
+			break;
+		}
+		case "info.sonicretro.org": {
+			// a full on wiki or index for the site
+			const unsupported = [
+				"Cite This Page - Sonic Retro",
+				"Compare pages - Sonic Retro",
+				"Export pages - Sonic Retro",
+				"What links here - Sonic Retro",
+				"Create account - Sonic Retro",
+				"Log in - Sonic Retro",
+				"Reset password - Sonic Retro",
+				"Login required - Sonic Retro",
+				"Preferences - Sonic Retro",
+				"Watchlist - Sonic Retro",
+				"Reset tokens - Sonic Retro",
+				"Blocked users - Sonic Retro",
+				"File list - Sonic Retro"
+			];
+			presenceData.smallImageKey = "info_512";
+			presenceData.smallImageText = "info.sonicretro.org";
+			if (
+				pathname === "/" ||
+				pathname.includes("/Main_Page" || "/Main_Page#")
+			) {
+				presenceData.largeImageKey = "info";
+				presenceData.details = "Zooming through the main page";
+			} else if (pathname.includes("Special:")) {
```

Comment:
this will never be true iirc

Reply:
Incorrect:

![image](https://user-images.githubusercontent.com/96930330/170850465-2fbb4771-6719-48d9-8ae1-d5e9c3cfb1c4.png)